File: src/main/webapp/tutorials/code/go/functions-variadic.go
package main

import "fmt"

// Variadic function (accepts variable number of arguments)
func sum(numbers ...int) int {
	total := 0
	for _, num := range numbers {
		total += num
	}
	return total
}

func printAll(prefix string, values ...interface{}) {
	fmt.Print(prefix, ": ")
	for _, v := range values {
		fmt.Print(v, " ")
	}
	fmt.Println()
}

func main() {
	fmt.Println("Sum:", sum(1, 2, 3))
	fmt.Println("Sum:", sum(10, 20, 30, 40, 50))
	
	// Passing a slice
	numbers := []int{1, 2, 3, 4, 5}
	fmt.Println("Sum:", sum(numbers...))
	
	printAll("Numbers", 1, 2, 3)
	printAll("Mixed", "Go", 42, true, 3.14)
}

File: src/main/webapp/tutorials/code/php/functions-variadic.php
<?php
// Variadic Functions

// Using ... (splat operator)
function sum(...$numbers)
{
    $total = 0;
    foreach ($numbers as $num) {
        $total += $num;
    }
    return $total;
}

echo sum(1, 2, 3);           // 6
echo "\n";
echo sum(1, 2, 3, 4, 5);     // 15
echo "\n";

// Variadic with regular parameters
function formatList($separator, ...$items)
{
    return implode($separator, $items);
}

echo formatList(", ", "apple", "banana", "cherry");
echo "\n";

// Type-hinted variadic
function sumIntegers(int ...$numbers): int
{
    return array_sum($numbers);
}

echo sumIntegers(10, 20, 30);  // 60
echo "\n";

// Unpacking arrays
$nums = [1, 2, 3, 4, 5];
echo sum(...$nums);  // 15
